### PR TITLE
Fix issue with delete request in `as_group`

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request

Add update `min_instance_number` before delete

## PR Checklist

* [x] Refers to: #529
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccASV1Group_basic
--- PASS: TestAccASV1Group_basic (96.70s)
=== RUN   TestAccASV1Group_RemoveWithSetMinNumber
--- PASS: TestAccASV1Group_RemoveWithSetMinNumber (150.02s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

File: opentelekomcloud/resource_opentelekomcloud_as_group_v1.go [hunk 1]
```diff
@@ -380,7 +335,7 @@ func refreshInstancesLifeStates(asClient *golangsdk.ServiceClient, groupID strin
 func checkASGroupInstancesInService(asClient *golangsdk.ServiceClient, groupID string, insNum int, timeout time.Duration) error {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{"PENDING"},
-		Target:  []string{"INSERVICE"}, //if there is no lifecyclestatus, meaning no instances in asg
+		Target:  []string{"INSERVICE"}, // if there is no LifeCycle status, meaning no instances in asg
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_as_group_v1.go]: `lifecycle` can be written without `PascalCase`

File: opentelekomcloud/validators.go [hunk 2]
```diff
@@ -399,3 +396,13 @@ func validateDDSStartTime(v interface{}, k string) (ws []string, errors []error)
 
 	return
 }
+
+func resourceASGroupValidateListenerId(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	split := strings.Split(value, ",")
+	if len(split) <= 3 {
+		return
+	}
```
Comment (outcatcher) [inline opentelekomcloud/validators.go]: `strings.Count(value, ',') == 3` should be a simplier solution for that

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

File: opentelekomcloud/validators.go [hunk 3]
```diff
@@ -399,3 +396,12 @@ func validateDDSStartTime(v interface{}, k string) (ws []string, errors []error)
 
 	return
 }
+
+func resourceASGroupValidateListenerId(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if strings.Count(value, ",") <= 3 {
```
Comment (outcatcher) [inline opentelekomcloud/validators.go]: Should be `2` not `3`

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by outcatcher — DISMISSED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by lego963** [comment]
recheck

**Pull request merged**